### PR TITLE
Feature/gh225 add additional details in stat

### PR DESCRIPTION
**Pull request by gin0115**
#### Changes proposed in this Pull Request

* Adds in a text message on the link report page which explains the current link states

#### Testing instructions

*

Mentions #225 

## Summary by CodeRabbit

* **New Features**
  * Dynamic link status summaries in reports with clearer archived vs non‑archived messages, correct pluralization, and redirect warnings.

* **Bug Fixes**
  * Broken-link detection now requires an archived copy to be present before counting as broken.

* **Performance**
  * Dashboard statistics cache shortened from 2 hours to 15 minutes for fresher data.

* **Improvements**
  * Check history displayed most-recent-first for readability.

* **Tests**
  * Added comprehensive unit tests covering summary messaging and edge cases.

### CODE REVIEW

File: src/Util/Link_Summary_Factory.php [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+		if ( null === $last_check ) {
+			return false;
+		}
+
+		$last_check = $this->link->get_last_check();
+		if ( ! is_array( $last_check ) || ! isset( $last_check['http_code'] ) ) {
+			return false;
+		}
+		return $this->is_http_code_valid( absint( $last_check['http_code'] ) );
+	}
```
[the remark this answers is not in the document]
Comment (gin0115) [reply src/Util/Link_Summary_Factory.php:120]: image
